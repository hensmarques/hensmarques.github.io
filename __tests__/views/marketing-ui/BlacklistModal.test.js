import React from 'react';
import BlacklistModal from '../../../src/views/marketing-ui/BlacklistModal';

describe('<BlacklistModal/>', () => {
  const mockOnClose = jest.fn();
  let translationSpy;
  let wrapper;

  beforeEach(() => {
    translationSpy = jest.spyOn(window, '_t').mockClear();
    wrapper = shallow(<BlacklistModal onClose={mockOnClose} />);
  });

  it('renders and matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Button').props().onClick).toBe(mockOnClose);
  });

  it('handles translations', () => {
    expect(translationSpy).toHaveBeenCalledWith(
      'Cointrader is not available in your country.',
      'BLACKLIST.MESSAGE'
    );
    expect(translationSpy).toHaveBeenCalledWith('OK', 'BLACKLIST.OK');
  });
});
