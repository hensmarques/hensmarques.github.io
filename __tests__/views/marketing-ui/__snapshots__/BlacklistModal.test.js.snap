// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlacklistModal/> renders and matches the snapshot 1`] = `
<Connect(Modal)
  className="blacklist-modal"
  onClose={[MockFunction]}
>
  <div
    className="modal-blacklist-content"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Cointrader is not available in your country.",
      }
    }
  />
  <div
    className="blacklist-button-container"
  >
    <Button
      className=""
      disabled={false}
      loading={false}
      loadingStatus={false}
      onClick={[MockFunction]}
      type="button"
    >
      OK
    </Button>
  </div>
</Connect(Modal)>
`;
